crypto.go

```diff
@@ -74,12 +79,17 @@ func (s dhkemScheme) getEphemeralKeyPair(rand io.Reader) (KEMPrivateKey, KEMPubl
 		return s.skE, s.skE.PublicKey(), nil
 	}
 
-	return s.group.GenerateKeyPair(rand)
+	ikm := make([]byte, s.PrivateKeySize())
+	rand.Read(ikm)
+
+	return s.group.DeriveKeyPair(ikm)
 }
 
 func (s dhkemScheme) extractAndExpand(dh []byte, kemContext []byte, Nzz int) []byte {
-	prk := s.KDF.LabeledExtract(nil, "dh", dh)
-	return s.KDF.LabeledExpand(prk, "prk", kemContext, Nzz)
+	idBuffer := make([]byte, 2)
+	binary.BigEndian.PutUint16(idBuffer, uint16(s.ID()))
+	eae_prk := s.group.internalKDF().LabeledExtract(nil, string(idBuffer)+"eae_prk", dh)
+	return s.group.internalKDF().LabeledExpand(eae_prk, "zz", kemContext, Nzz)
```

Comment:
The spec has KEM_ID || "zz" as the label for LabeledExpand. I think that's an error in the spec, though. No other LabeledExpand calls have KEM binding. When KEM binding is necessary, it's done in LabeledExtract.